src/iqm_client/iqm_client.py

```diff
@@ -349,21 +348,20 @@ class IQMClient:
     def __init__(
             self,
             url: str,
-            settings: Optional[dict[str, Any]] = None,
             **credentials  # contains auth_server_url, username, password
     ):
         if not url.startswith(('http:', 'https:')):
             raise ClientConfigurationError(f'The URL schema has to be http or https. Incorrect schema in URL: {url}')
         self._base_url = url
-        self._settings = settings
         self._credentials = _get_credentials(credentials)
         self._update_tokens()
 
     def submit_circuit(
             self,
             circuit: Circuit,
             qubit_mapping: Optional[list[SingleQubitMapping]] = None,
```

Comment:
While we are changing the API, I would suggest to make the optional stuff keyword-only, for clarity.
```suggestion
            *,
            qubit_mapping: Optional[list[SingleQubitMapping]] = None,
```

Reply:
This way we can rearrange them without breaking the API, and make user errors harder to make.

Reply:
This is a good idea. Another parameter will be added in the next minor version already (`calibration_set_id`).

Reply:
Done!